packages/selfcare-client-users-updater-consumer/test/selfcareClientUsersUpdaterProcessor.test.ts

```diff
@@ -0,0 +1,169 @@
+/* eslint-disable prefer-const */
+/* eslint-disable functional/no-let */
+import {
+  vi,
+  afterEach,
+  beforeAll,
+  beforeEach,
+  describe,
+  MockInstance,
+  it,
+  expect,
+} from "vitest";
+import {
+  InteropTokenGenerator,
+  RefreshableInteropToken,
+  userRole,
+} from "pagopa-interop-commons";
+import { EachMessagePayload } from "kafkajs";
+import { selfcareClientUsersUpdaterProcessorBuilder } from "../src/services/selfcareClientUsersUpdaterProcessor.js";
+import { config } from "../src/config/config.js";
+import {
+  AuthorizationProcessClient,
+  authorizationProcessClientBuilder,
+} from "../src/clients/authorizationProcessClient.js";
+import {
+  allowedOriginsUuid,
+  correctEventPayload,
+  generateInternalTokenMock,
+  kafkaMessagePayload,
+} from "./utils.js";
+
+describe("selfcareClientUsersUpdaterProcessor", () => {
+  let authorizationProcessClientMock: AuthorizationProcessClient =
+    authorizationProcessClientBuilder(config.authorizationProcessUrl);
+  let tokenGeneratorMock = new InteropTokenGenerator(config);
+  let refreshableTokenMock = new RefreshableInteropToken(tokenGeneratorMock);
+  let selfcareClientUsersUpdaterProcessor: ReturnType<
+    typeof selfcareClientUsersUpdaterProcessorBuilder
+  >;
+
+  beforeAll(async () => {
+    selfcareClientUsersUpdaterProcessor =
+      selfcareClientUsersUpdaterProcessorBuilder(
+        refreshableTokenMock,
+        authorizationProcessClientMock,
+        config.interopProductId,
+        allowedOriginsUuid
+      );
+  });
+
+  let refreshableInternalTokenSpy: MockInstance;
+  // todo manca la spy della chiamata verso auth-process
```

Comment:
🇬🇧 , but also if you wants to test success case you can track this as Jira task to handle later or you can try to: 
-  selfcareClientUsersUpdaterProcessor.ts : create a dummy exported function called `removeAdminClient()` use it in a success case
- in this test spy exported function and verify it was called

when the api clients was updated you replace it with properly mocked client.